assets/js/theme/common/carousel/index.js

```diff
@@ -2,62 +2,61 @@ import 'slick-carousel';
-    slideCount, $slides, options: { slidesToShow, slidesToScroll },
-}) => getActiveSlideIdxAndSlidesQuantity(
-    slideCount,
-    slidesToShow,
-    slidesToScroll,
-    $slides,
-);
+export const setCarouselState = ({ delegateTarget }, carouselObj) => {
+    const carouselObjCurrent = carouselObj || delegateTarget.slick;
+    const { $slider } = carouselObjCurrent;
+
+    $slider.data('state', getActiveSlideIdxAndSlidesQuantity(carouselObjCurrent));
+};
 
 export const onUserCarouselChange = ({ data }, context, $slider) => {
     const $activeSlider = $slider || data;
-    const $parentContainer = $activeSlider.hasClass('productView-thumbnails') ? $('.productView-images') : $activeSlider;
-    const { activeSlideIdx, slidesQuantity } = extractSlidesDetails($activeSlider[0].slick);
+    const $parentContainer = $activeSlider.hasClass('productView-thumbnails') ? $activeSlider.parent('.productView-images') : $activeSlider;
+    const { activeSlideIdx, slidesQuantity } = $activeSlider.data('state');
     const $carouselContentElement = $('[data-carousel-content-change-message]', $parentContainer);
     const carouselContentAnnounceMessage = updateTextWithLiveData(context.carouselContentAnnounceMessage, (activeSlideIdx + 1), slidesQuantity);
 
     $carouselContentElement.text(carouselContentAnnounceMessage);
 };
 
-export const onSlickCarouselChange = (e, carousel, context) => {
+export const onSlickCarouselChange = (e, carouselObj, context) => {
     const {
         $dots,
         $slider,
         $prevArrow,
         $nextArrow,
-    } = carousel;
+        options: { infinite },
+    } = carouselObj;
 
-    const { activeSlideIdx, slidesQuantity } = extractSlidesDetails(carousel);
+    const { activeSlideIdx, slidesQuantity } = $slider.data('state') || getActiveSlideIdxAndSlidesQuantity(carouselObj);
 
     dotsSetup($dots, activeSlideIdx, slidesQuantity, context);
-    arrowAriaLabling($prevArrow, $nextArrow, activeSlideIdx, slidesQuantity, context.carouselArrowAndDotAriaLabel);
-    setTabindexes($slider.find('.slick-slide'));
-    tooltipSetup($prevArrow, $nextArrow, $dots);
-    activatePlayPauseButton(carousel, slidesQuantity, context);
+    arrowAriaLabling($prevArrow, $nextArrow, activeSlideIdx, slidesQuantity, infinite, context.carouselArrowAndDotAriaLabel);
+    analizeSlides($slider.find('.slick-slide'));
+    refreshFocus($prevArrow, $nextArrow, $dots, $slider, activeSlideIdx, slidesQuantity, infinite);
+
+    $slider.data('state', null);
 };
 
 export default function (context) {
     $('[data-slick]').each((idx, carousel) => {
         // getting element using find to pass jest test
         const $carousel = $(document).find(carousel);
+
+        $carousel.on('init breakpoint swipe', setCarouselState);
+        $carousel.on('click', '.slick-arrow, .slick-dots', $carousel, setCarouselState);
+
+        $carousel.on('init breakpoint', (e, carouselObj) => activatePlayPauseButton(e, carouselObj, context));
```

Comment:
as a suggestion: If attached video doesn't contain breakpoint event maybe it's worth to add it for demonstration purpose.

Reply:
It was added previously for scalability - if any developer or merchant would like to add play pause button to product carousels they will spend 5mins - on adding button component to carousel and update slick option of autoplay to true. Now just performance improvements. Main purpose of breakpoint here - update the order of carousel controls, because slick regenerates them on breakpoint 

https://user-images.githubusercontent.com/66325265/114515204-c6af0480-9c44-11eb-8cf2-7d9e40ffd9c1.mov